client/asset/btc/btc.go

```diff
@@ -5792,3 +5797,82 @@ func deserializeMsgTx(r io.Reader) (*wire.MsgTx, error) {
 func msgTxFromBytes(txB []byte) (*wire.MsgTx, error) {
 	return deserializeMsgTx(bytes.NewReader(txB))
 }
+
+// ConfirmRedemption returns how many confirmations a redemption has. Normally
+// this is very straitforward. However, with fluxuating fees, there's the
+// possibility that the tx is never mined and eventually purged from the
+// mempool. In that case we use the provided fee suggestion to create and send
+// a new redeem transaction, returning the new transactions hash.
+func (btc *baseWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption, feeSuggestion uint64) (*asset.ConfirmRedemptionStatus, error) {
+	txHash, _, err := decodeCoinID(coinID)
+	if err != nil {
+		return nil, err
+	}
+
+	_, confs, err := btc.rawWalletTx(txHash)
+	// redemption transaction found, return its confirms.
+	//
+	// TODO: Investigate the case where this redeem has been sitting in the
+	// mempool for a long amount of time, possibly requiring some action by
+	// us to get it unstuck.
+	if err == nil {
+		return &asset.ConfirmRedemptionStatus{
+			Confs:  uint64(confs),
+			Req:    requiredRedeemConfirms,
+			CoinID: coinID,
+		}, nil
+	}
+
+	if !errors.Is(err, WalletTransactionNotFound) {
+		return nil, fmt.Errorf("problem searching for redemption transaction %s: %w", txHash, err)
+	}
+
+	// Redemption transaction is missing from the point of view of our node!
+	// Unlikely, but possible it was redeemed by another transaction. Check
+	// if the contract is still an unspent output.
```

Comment:
For both DCR and BTC, in the case where the wallet/node gives a result for the transaction, and we declare "mempool for a long amount of time" (the case from about 15 lines above), it could jump down here to attempt to locate a different spender.

If it found a different spender, that would really be indicative of a wallet bug because the wallet should automatically abandon/remove a transaction if its inputs are spent, but it's something we might want to try in the future, for robustness.

Regarding doing an explicit `AbandonTransaction` (with DCR) or otherwise attempting to replace the original, I still maintain that is dangerous because there is a chance that:

1. It is still on the network waiting to be mined, and any replacement may be rejected by the network on broadcast
2. The outputs of the original redeem may be spent, and abandoning/replacing the redeem will have adverse consequences for those 0-conf spenders
3. The redeem may **actually be mined** and our node/wallet could just have lost connection or gotten stuck or corrupted or whatever, and we're falsely concluding it's not mined.  We could be creating tons of unnecessary issues for ourselves if we start abanonding, replacing, trying to bump with cpfp, etc.  A restart or rescan of the wallet may very well be the best and only real resolution.